app/src/main/java/org/kiwix/kiwixmobile/settings/CustomSwitchPreference.java

```diff
@@ -64,6 +65,8 @@ protected void onBindView(View view) {
     ViewGroup viewGroup = (ViewGroup) view;
     clearListenerInViewGroup(viewGroup);
     super.onBindView(view);
+    SharedPreferenceUtil sharedPreferenceUtil = new SharedPreferenceUtil(getContext());
+    sharedPreferenceUtil.disableNightMode(view);
```

Comment:
This is not the responsibility of the SharedPrefernceUtil.
Can you pursue a solution where you listen to changes in KiwixSettingsActivity? See KiwixSettingsActivity.onSharedPreferenceChanged

Reply:
Yes, I had initially planned on doing that, but I read @mhutti1 's review on a [PR](https://github.com/kiwix/kiwix-android/pull/1073), in which he had said we should be using SharedPrefernceUtil for this functionality. 
I will make the required changes.